dist-build/android-build.sh

```diff
@@ -18,18 +18,23 @@ if [ ! -f ./configure ]; then
   exit 1
 fi
 
-if [ -z "$TARGET_ARCH" ] || [ -z "$ARCH" ] || [ -z "$HOST_COMPILER" ]; then
+if [ "x$HOST_COMPILER" = 'x' ]; then
   echo "You shouldn't use android-build.sh directly, use android-[arch].sh instead" >&2
   exit 1
 fi
 
-export PREFIX="$(pwd)/libsodium-android-${TARGET_ARCH}"
-export TOOLCHAIN_OS_DIR="$(uname | tr '[:upper:]' '[:lower:]')-x86_64/"
+if [ -z ${PREFIX+x} ]; then export PREFIX="$(pwd)/libsodium-android"; echo "setting default PREFIX=${PREFIX}"; fi
+if [ -z ${LIB_PATH_PREFIX+x} ]; then echo ""; else export LIB_PATH="${LIB_PATH_PREFIX}/${TARGET_ARCH_DIR}"; fi
+if [ -z ${LIB_PATH+x} ]; then export LIB_PATH="${PREFIX}/lib/${TARGET_ARCH_DIR}"; fi
+if [ -z ${INCLUDE_PATH+x} ]; then export INCLUDE_PATH="${PREFIX}/include"; fi
+if [ -z ${TARGET_ARCH_DIR+x} ]; then export TARGET_ARCH_DIR="${HOST_COMPILER}"; fi
+
+export TOOLCHAIN_OS_DIR="`uname | tr '[:upper:]' '[:lower:]'`-x86_64/"
 export TOOLCHAIN_DIR="$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/${TOOLCHAIN_OS_DIR}"
 echo "$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/${TOOLCHAIN_OS_DIR}/${HOST_COMPILER}"
 
 export PATH="${PATH}:${TOOLCHAIN_DIR}/bin"
-SDK_VERSION_NUM=$(echo $NDK_PLATFORM | cut -d'-' -f2)
+SDK_VERSION_NUM=`echo $NDK_PLATFORM | cut -d'-' -f2`
```

Comment:
Modern syntax is preferred. I'd highly recommend using the `shellcheck` tool to check the script against common errors, including obsolete constructions. There's also a Visual Studio Code plugin for it. It's a great tool!

Reply:
sorry never heard of shellcheck I'll run my scripts through it before any more commits.